src/arfsdao.ts

```diff
@@ -490,10 +544,12 @@ export class ArFSDAO extends ArFSDAOAnonymous {
 		const unixTime = Math.round(Date.now() / 1000);
 
 		// Gather file information
-		const fileStats = fs.statSync(filePath);
-		const fileData = fs.readFileSync(filePath);
-		const dataContentType = extToMime(filePath);
-		const lastModifiedDateMS = Math.floor(fileStats.mtimeMs);
+		const fileStats = wrappedFile.fileStats;
```

Comment:
These all use the wrappedFile class, consider a single gather file information method on the file wrapper for both public and private methods to share